src/main/java/uk/gov/pay/directdebit/common/exception/ApiException.java

```diff
@@ -0,0 +1,19 @@
+package uk.gov.pay.directdebit.common.exception;
+
+import org.slf4j.Logger;
+import uk.gov.pay.directdebit.app.logger.PayLoggerFactory;
+
+public class ApiException extends RuntimeException {
+    private static final Logger LOGGER = PayLoggerFactory.getLogger(ApiException.class);
+    private Integer statusCode;
+
+    Integer getStatusCode() {
+        return statusCode;
+    }
+
+    ApiException(Integer statusCode, String message) {
+        super(message);
+        this.statusCode = statusCode;
+        LOGGER.error(message);
```

Comment:
I would log the error wherever it happens. Logging is easier to read since it appears where in the class was caused rather than here (I know it saves lines of code in places but we lose the point of logging where the thing happened). Logs are logged as Whatever.class "Error message happened here"